src/main/main.js

```diff
@@ -593,17 +592,52 @@ function handleAppWebContentsCreated(dc, contents) {
   contents.on('new-window', (event, url) => {
     const parsedURL = urlUtils.parseURL(url);
 
+    // Dev tools case
     if (parsedURL.protocol === 'devtools:') {
       return;
     }
     event.preventDefault();
 
+    // Check for valid URL
+    if (!urlUtils.isValidURI(url)) {
+      return;
+    }
+
+    // Check for custom protocol
+    if (parsedURL.protocol !== 'http:' && parsedURL.protocol !== 'https:' && parsedURL.protocol !== `${scheme}:`) {
+      allowProtocolDialog.handleDialogEvent(parsedURL.protocol, url);
+      return;
+    }
+
     const server = urlUtils.getServer(parsedURL, config.teams);
 
     if (!server) {
       shell.openExternal(url);
       return;
     }
+
+    // Public download links case
+    // TODO: This doesn't work correctly right now, needs to be refactored
```

Comment:
@Willyfrog Realized this wasn't migrated over either, it's meant to handle public links to files hosted on your MM server. But it doesn't seem to be working correctly, complaining about the URL type we're using. Something we should verify how we want to handle and hopefully fix the issue.

Funny enough it doesn't really work in the current desktop app because we don't use the same path that matched the original regex anymore, so we currently just show the image in the same `webview` and you have to press Back to get out of it.

Reply:
is this related to https://mattermost.atlassian.net/browse/MM-32333 ?